check_domain_access.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+/// The script checks if the domains accessed by the pub client are available,
+/// and not blocked by any network infrastructure.
+Future<void> main(List<String> args) async {
+  print('Testing host resolves...');
+  await _checkHosts();
+
+  print('Testing one connection per request...');
+  await _checkUrls();
+
+  print('Testing connection reuse...');
+  final iterations = args.isEmpty ? 10 : int.parse(args.first);
+  final client = HttpClient();
+  for (var i = 0; i < iterations; i++) {
+    print('- iteration #${i + 1}/$iterations...');
+    await _checkUrls(client: client);
+  }
+  client.close();
+
+  print('Done.');
+  exit(0);
+}
+
+final timeLimit = Duration(seconds: 15);
```

Comment:
`l`: Is this some random number, or did you choose this with intention? If the latter, we could add a comment about the reasoning.